tests/layer_tests/tensorflow_tests/test_tf_MatMul.py

```diff
@@ -0,0 +1,97 @@
+# Copyright (C) 2022 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+
+import numpy as np
+import pytest
+from common.tf_layer_test_class import CommonTFLayerTest
+
+
+class TestMatMul(CommonTFLayerTest):
+
+    def create_net_with_matmul_op(self, x_shape, y_shape, x_bool, y_bool, op_type, ir_version, use_new_frontend):
+        """
+            Tensorflow net                 IR net
+
+            Input->op_type        =>       Input->MatMul
+                     /                              /
+            Const---/                      Const---/
+        """
+        import tensorflow as tf
+        op_type_to_tf = {
+            'BatchMatMul': tf.raw_ops.BatchMatMul,
+            'BatchMatMulV2': tf.raw_ops.BatchMatMulV2,
+            'BatchMatMulV3': tf.raw_ops.BatchMatMulV3,
+            'MatMul': tf.raw_ops.MatMul,
+        }
+
+        tf.compat.v1.reset_default_graph()
+
+        # Create the graph and model
+        with tf.compat.v1.Session() as sess:
+            tf_x_shape = x_shape.copy()
+            tf_y_shape = y_shape.copy()
+
+            tf_x = tf.compat.v1.placeholder(tf.float32, tf_x_shape, 'Input')
+            constant_value = np.random.randint(-3, 3, tf_x_shape).astype(np.float32)
+            tf_y = tf.constant(constant_value)
+            if op_type == 'MatMul':
+                if len(x_shape) > 2 or len(y_shape) > 2:
+                    pytest.skip("MatMul doesn't support rank > 2")
+                op_type_to_tf[op_type](a=tf_x, b=tf_y, transpose_a=x_bool, transpose_b=y_bool, name='Operation')
+            elif op_type == 'BatchMatMulV3':
+                op_type_to_tf[op_type](x=tf_x, y=tf_y, Tout=tf.float32, adj_x=x_bool, adj_y=y_bool, name='Operation')
+            else:
+                op_type_to_tf[op_type](x=tf_x, y=tf_y, adj_x=x_bool, adj_y=y_bool, name='Operation')
+
+            tf.compat.v1.global_variables_initializer()
+            tf_net = sess.graph_def
+
+        ref_net = None
+
+        return tf_net, ref_net
+
+    test_data_precommit = [
+        dict(x_shape=[4, 4], y_shape=[4, 4]),               #Tests 2D shapes
+        dict(x_shape=[4, 4, 4, 4], y_shape=[4, 4, 4, 4]),   #Tests 4D shapes
```

Comment:
Let us have more interesting cases with different dimension like: `[2,3,4,5] [2,3,5,6]`, etc. In this case you need to specify `adj_x` and `adj_y` along with input shapes

Reply:
In precommit I cover only simplest cases. Cases with higher complexity I suggest to add to the nightly, what do you think?
Especially because for precommit adj_x and adj_y forced to default-False value.

Reply:
Moreover, in case of real usage of adj_x, adj_y and transpose_a, transpose_b - we need much more tests. And I think it isn't a target of current stage.